automation/src/test/scala/org/broadinstitute/dsde/test/api/SamApiSpec.scala

```diff
@@ -255,6 +255,19 @@ class SamApiSpec extends FreeSpec with BillingFixtures with Matchers with ScalaF
         petSaKeyIdOriginal should not equal petSaKeyIdNew //make sure we were able to generate a new key and that a new one was returned
       }
     }
+
+    "should get an access token for a user's own pet service account" in {
+      val user = UserPool.chooseStudent
+
+      withCleanBillingProject(UserPool.chooseProjectOwner, List(user.email)) { project =>
+
+        val token1 = Sam.user.petServiceAccountToken(project)(user.makeAuthToken)
+
+        val token2 = Sam.user.petServiceAccountToken(project)(user.makeAuthToken)
+
+        token1 shouldNot be(token2)
```

Comment:
Does the token necessarily change for subsequent calls to `petServiceAccountToken` from the same user? I could see a service returning the same token for a user until it expires (the Google [javascript API](https://developers.google.com/api-client-library/javascript/reference/referencedocs) does this for example).

A cool test would be to _use_ the token to make a call as the pet, and verify that it succeeds. Would this test be valid?
```
// get my pet's email
val petEmail1 =  Sam.user.petServiceAccountEmail(project)(user.makeAuthToken)
// get my pet's token
val petToken = Sam.user.petServiceAccountToken(project)(user.makeAuthToken)
// get my pet's email using my pet's token
val petEmail2 = Sam.user.petServiceAccountEmail(project)(petToken)
// result should be the same
petEmail2 shouldBe petEmail1
```